assignment-1/Functions.java

```diff
@@ -0,0 +1,271 @@
+                    .collect(Collectors.toList());
+            for (Players lp : topBallers) {
+                System.out.println(lp.getName());
+            }
+            List<Players> topAllrounders = allPlayers.stream()
+                    .sorted(Comparator.comparingInt(Players::getAllRounders).reversed()).limit(3)
+                    .collect(Collectors.toList());
+            System.out.println("\nTop 3 AllRounders of the season");
+            for (Players lp : topAllrounders) {
+                System.out.println(lp.getName());
+            }
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static void highestRunScorerTeam() {
+        HashMap<String, Double> teamAndAvg = new HashMap<>();
+        LinkedHashMap<String, Double> reverseSorted = new LinkedHashMap<>();
+        try {
+            for (Team t : allTeams) {
+                List<Players> topBatsman = t.players.stream()
+                        .sorted(Comparator.comparingDouble(Players::getAverage).reversed()).limit(6)
+                        .collect(Collectors.toList());
+                Double score = topBatsman.stream().mapToDouble(Players::getAverage).sum();
+                teamAndAvg.put(t.Name, score);
+            }
+            teamAndAvg.entrySet()
+                    .stream()
+                    .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder())).limit(2)
+                    .forEachOrdered(x -> reverseSorted.put(x.getKey(), x.getValue()));
+            Set<String> keys = reverseSorted.keySet();
+            System.out.println("Task 7");
+            System.out.println("Below are two Teams which would score the highest runs");
+            System.out.println("________________________________________________________");
+            System.out.println("Team | Predicted Score");
+
+            for (String key : keys) {
+                System.out.println(key + "  | " + Math.round(reverseSorted.get(key)));
+            }
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        } finally {
+            System.out.println("________________________________________________________");
+        }
+    }
+
+
+    // Eighth Task
+    public static void nextGenPlayers() {
+        Set<Players> nextGenPlayers = new HashSet<Players>();
+        try {
+
+            for (Players newGenPlayer : allPlayers) {
+                if (newGenPlayer.getMatches() <= 20) {
+                    if (newGenPlayer.getAverage() >= 15.0) {
+                        nextGenPlayers.add(newGenPlayer);
+                    }
+                }
+                if (newGenPlayer.getMatches() <= 15) {
+                    if (newGenPlayer.getWickets() >= 10) {
+                        nextGenPlayers.add(newGenPlayer);
+                    }
+                }
+            }
+            for (Players newPlayers : nextGenPlayers) {
```

Comment:
This feels like an incomplete task, since the question was asked to find nexgen players for all teams,
I do not see any team names being printed